src/routers/v2/DelaysRouter.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import ApplicationRouter from '../../appdev/ApplicationRouter';
+import RealtimeFeedUtils from '../../utils/RealtimeFeedUtils';
+
+class DelaysRouter extends ApplicationRouter<Object> {
+  constructor() {
+    super(['POST']);
+  }
+
+  getPath(): string {
+    return '/delays/';
+  }
+
+  async content(req): Promise<any> {
+    const delays = req.body.data;
+    const rtf = await RealtimeFeedUtils.fetchRTF();
+    delays.map(async (stopAndTripID) => {
+      const res = await RealtimeFeedUtils.getDelayInformation(
+        stopAndTripID.stopID, stopAndTripID.tripID, rtf,
+      );
+      stopAndTripID.delay = res ? res.delay : null;
+    });
+    return Promise.all(delays);
```

Comment:
something seems wrong here -- you throw away the returned `Promise<void>`s from this map. this might still work (maybe the mutation makes it fine?) but i feel like this should really just be immutable and read something like 
```
return {
  ...stopAndTripID,
  delay: res ? res.delay : null,
}
```
in the map callback and then you'd return the `Promise.all` of the map result rather than the original `delays`.

Reply:
i'm confused - what's wrong with modifying the array?

Reply:
so my main concern is that `delays` is not a promise or array of promises, making your `Promise.all` call invalid, and also that the return value of the `map` is just not being used. i think the typing issues aren't being properly caught here -- it's an example of a type-unsound but functional program

Reply:
ahhh yes @kq-li is right, you should return the `map` value here, I thought it was in-place but `map` returns a new array!